### PR TITLE
SCALE-243 Configure Ruby 3 hash style, update Github workflow

### PR DESCRIPTION
**Pull request by honigc**
https://panoramaed.atlassian.net/browse/SCALE-243

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -4,20 +4,21 @@ jobs:
...
       with:
-        ruby-version: 2.7
-    - uses: actions/cache@v1
-      with:
-        path: vendor/bundle
-        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-gems-
-    - run: bundle config path vendor/bundle
-    - run: bundle install --jobs 4 --retry 3
```
Comment (parkerfinch) [inline .github/workflows/tests.yml:21]: Why can these be removed? Does bundler do this automatically?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -4,20 +4,21 @@ jobs:
...
     - uses: actions/checkout@master
     - uses: ruby/setup-ruby@v1
       with:
-        ruby-version: 2.7
-    - uses: actions/cache@v1
-      with:
-        path: vendor/bundle
-        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-gems-
```
Comment (parkerfinch) [inline .github/workflows/tests.yml:19]: Is this `actions/cache@v1` section replaced by the `bundler-cache: true`?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (honigc) [reply .github/workflows/tests.yml:21]: I think this happens automatically in `ruby/setup-ruby` now.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (honigc) [reply .github/workflows/tests.yml:19]: I believe so. I basically copied these changes over from [memo_wise](https://github.com/panorama-ed/memo_wise/blob/main/.github/workflows/main.yml#L29), without the comments.

Comment (parkerfinch) [reply .github/workflows/tests.yml:19]: Ah yeah looks like this is a typical change: https://www.jessesquires.com/blog/2021/08/23/caching-bundler-on-github-actions/

**Review by parkerfinch — APPROVED** [review]

**Pull request merged**